### PR TITLE
ref(types): Fix return type on base request parser

### PR DESCRIPTION
**Pull request by leeandher**
Fixes mypy errors for base request parser.

### CODE REVIEW

**Review by dashed — APPROVED** [review]

File: pyproject.toml
Comment (leeandher) [inline pyproject.toml:1]: Hoping this is the correct way to remove a file from the blocklist 🤞 let me know if there's a better way

File: pyproject.toml [hunk 1]
```diff
@@ -46,7 +46,11 @@ looponfailroots = ["src", "tests"]
 [tool.mypy]
 python_version = "3.8"
 mypy_path = ["fixtures/stubs-for-mypy"]
-plugins = ["pydantic.mypy", "mypy_django_plugin.main", "tools.mypy_helpers.plugin"]
+plugins = [
+    "pydantic.mypy",
+    "mypy_django_plugin.main",
+    "tools.mypy_helpers.plugin",
+]
```
Comment (asottile-sentry) [inline pyproject.toml:53]: this part is an unrelated change -- please revert this if possible!

**Review by asottile-sentry — APPROVED** [review]

**Comment by asottile-sentry** [comment]
looks like a few more will need the HttpResponse -> HttpResponseBase treatment:

```
src/sentry/middleware/integrations/parsers/vsts.py:34: error: Incompatible return value type (got "HttpResponseBase", expected "HttpResponse")  [return-value]
src/sentry/middleware/integrations/parsers/vsts.py:39: error: Incompatible return value type (got "HttpResponseBase", expected "HttpResponse")  [return-value]
src/sentry/middleware/integrations/parsers/msteams.py:34: error: Incompatible return value type (got "HttpResponseBase", expected "HttpResponse")  [return-value]
src/sentry/middleware/integrations/parsers/msteams.py:39: error: Incompatible return value type (got "HttpResponseBase", expected "HttpResponse")  [return-value]
src/sentry/middleware/integrations/parsers/gitlab.py:79: error: Incompatible return value type (got "HttpResponseBase", expected "HttpResponse")  [return-value]
src/sentry/middleware/integrations/parsers/bitbucket_server.py:21: error: Incompatible return value type (got "HttpResponseBase", expected "HttpResponse")  [return-value]
```

**Review by asottile-sentry — APPROVED** [review]
